### DISCUSSION TITLE
Agile Negative Rate Handling + Standing Charge in Daily Cost

### DISCUSSION THREAD

**Discussion by anothermort** [BottlecapDave/HomeAssistant-OctopusEnergy#412] [Q&A] [answered]
Hi

Keeping an eye on Agile rates and usage early this morning, as we had a plunge the electric cost was going down as we use more.
As expected, but great to see it in near real time using the integration and the mini.

What happens if the cost goes negative?

1.  Does the standing charge go if you have used enough energy at negative prices to cancel it out.
2.  Or will you always have the standing charge shown by the integration?

I hope that makes sense.

Just have only gone negative before we had the mini and trying to understand the expected result. So I can alter the display software to handle it correctly.

As always thanks for the great integration it was nice to watch the daily cost count down from early in the morning usage.

**Comment by BottlecapDave** [comment, marked as answer]
Hello. I assume you mean regarding the `current_accumulative_cost` sensors? If so, then the main value will go negative if you use more in the negative rate than you consume in the positive plus the standing charge. 

Example

```
5kwh at -10p
1kwh at 10p
standing charge at 30p

total positive = 10p
total negative = -50p
total without standing charge = -40p
total with standing charge = -10p
```

There is also an attribute if you want the value without the standing charge if that's clearer.

**Reply by anothermort** [reply]
Yes it is the  current_accumulative_cost sensors
Thankyou that makes complete sense.
